### PR TITLE
Move functionality from "./frescobaldi" to "./frescobaldi_app/__main__.py"

### PR DESCRIPTION
**Pull request by jeanas**
This is split out from #1597.

We want to move away from using setuptools' `data_files` argument, which is deprecated. But we still want this functionality, for installing the desktop files, the icon and the man page.

In setuptools, there is no non-deprecated replacement for it. On the other hand, Hatch supports an equivalent. However, due to the way Hatch autodiscovers the files that should be put into wheels, we need the "frescobaldi_app" directory renamed to "frescobaldi" (to match the overall project's name), which clashes with the existing "./frescobaldi" script.

This moves the functionality from that script to `frescobaldi_app/__main__.py` and reduces its use; the script itself will be removed in a separate PR.

The new way to run Frescobaldi from source without installing is `python -m frescobaldi_app` (soon to become `python -m frescobaldi`).

### CODE REVIEW

**Comment by dliessi** [comment]
I'm going to test on Mac later tonight.
I already see a problem: the new version of both the Mac and the Windows scripts is looking for `frescobaldi/__main__.py` instead of `frescobaldi_app/__main__.py`.
I'll fix it locally and test both runnign from source and building the application bundle.

**Comment by fedelibre** [comment]
I managed to run your main branch from source on Windows! (well, I don't have python-poppler-qt5 installed of course! but it's not a hard dependency)

I thought it was difficult, but it was pretty easy actually. Thanks to the instructions in #1597

File: macosx/mac-app.py [hunk 1]
```diff
@@ -49,7 +49,7 @@
 parser.add_argument('-s', '--script', \
   help = 'path of {0}\'s main script; you should use an absolute path, \
   so that the application bundle can be moved to another \
-  directory'.format(appinfo.appname), default = '{0}/{1}'.format(root, appinfo.name))
+  directory'.format(appinfo.appname), default = '{0}/{1}/__main__.py'.format(root, appinfo.name))
```
Comment (dliessi) [inline macosx/mac-app.py:52]: In this PR you do not rename the `frescobaldi_app` directory, so this should be `default = '{0}/{1}_app/__main__.py'.format(root, appinfo.name)`.
I made this change locally and tested building the application bundle and it worked.

File: windows/freeze.py [hunk 2]
```diff
@@ -84,7 +84,7 @@
 shutil.rmtree(target_dir, ignore_errors = True)
 
 frescobaldi = Executable(
-    '../frescobaldi',
+    '../frescobaldi/__main__.py',
```
Comment (dliessi) [inline windows/freeze.py:87]: This should be `'../frescobaldi_app/__main__.py'`.

File: windows/frescobaldi-wininst.py [hunk 3]
```diff
@@ -41,7 +41,7 @@
 
 
 # Main application script and Python executable:
-script = os.path.join(sys.prefix, 'Scripts', 'frescobaldi')
+script = os.path.join(sys.prefix, 'Scripts', 'frescobaldi', '__main__.py')
```
Comment (dliessi) [inline windows/frescobaldi-wininst.py:44]: I believe this should be `os.path.join(sys.prefix, 'Scripts', 'frescobaldi_app', '__main__.py')`, but I'm not sure (where does `Scripts` come from?) and cannot test on Windows now.

**Review by dliessi — CHANGES_REQUESTED** [review]
Running from source on Mac seems to work correctly, as does building the application bundle after applying the change I request on `macosx/mac-app.py`.

**Review by dliessi — APPROVED** [review]

**Comment by fedelibre** [comment]
I'm fine with going ahead anyway. We can fix any bug later.

I've always wanted to test the Windows installer and I have a Windows laptop.. but no time at the moment. I will have some time for this in a couple of weeks. But I wouldn't hold this PR until then.

**Pull request merged**